contracts/TokenListManager.sol

```diff
@@ -6,176 +6,80 @@ import "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
 import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
 import "@openzeppelin/contracts/token/ERC721/IERC721.sol";
 import "@openzeppelin/contracts/token/ERC1155/IERC1155.sol";
-import "@openzeppelin/contracts/utils/introspection/IERC165.sol";
-
-contract TokenListManager is ITokenListManager {
-
-    // Storage the token address to whitelist or blacklist
-    address[] private whitelistedTokensArray;
-    address[] private blacklistedTokensArray;
-    // Declare arrays to store addresses of whitelisted tokens by type
-    address[] private whitelistErc20Tokens;
-    address[] private whitelistErc721Tokens;
-    address[] private whitelistErc1155Tokens;
-    // Declare arrays to store addresses of blacklisted tokens by type
-    address[] private blacklistErc20Tokens;
-    address[] private blacklistErc721Tokens;
-    address[] private blacklistErc1155Tokens;
-
-    // Use mapping to check if token in whitelist
-    mapping(address => bool) private whitelistedTokens;
-    // Use mapping to check if token in blacklist
-    mapping(address => bool) private blacklistedTokens;
-
-    // Set permissions
-    modifier onlyOwner() {
-        // Match address
-        /** 0xC0Ec75c25EC201885a791Fd8d39Bf8CE96e1c566
-         * this address is my own test address
-         * owner can use thier own address to replace. */
-        require(msg.sender == 0xC0Ec75c25EC201885a791Fd8d39Bf8CE96e1c566, "Not authorized");
-        _;
+import "@openzeppelin/contracts/access/AccessControl.sol";
+
+contract TokenListManager is ITokenListManager, AccessControl {
+
+    // Utilize EnumerableSet for efficient set operations.
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    // Define roles for administrative control and user access.
+    bytes32 public constant ADMIN_ROLE = keccak256("ADMIN_ROLE");
+    bytes32 public constant USER_ROLE = keccak256("USER_ROLE");
```

Comment:
It seems not used?

Reply:
It is not used in this code for now. I thought there might be a need for user permissions in the future, so I wrote it in advance.